examples/main.tf

```diff
@@ -59,21 +59,60 @@ resource "docker_access_token" "access_token" {
   scopes      = local.token_scopes
 }
 
+# === Repository Tags Data Source Example ===
+# This demonstrates the key security use case: converting human-friendly tags to digest-pinned references
 
-# Output Demos
+# Get Alpine tags for our deployment
+data "docker_hub_repository_tags" "example" {
+  namespace = "library"
+  name      = "alpine"
+}
+
+# Get digest-pinned reference
+locals {
+  secure_image = "alpine@${data.docker_hub_repository_tags.example.tags["latest"].digest}"
+}
+
+# === Outputs ===
+
+# Repository outputs
 output "repo_output" {
-  value = resource.docker_hub_repository.org_hub_repo
+  description = "Created repository information"
+  value = {
+    id          = docker_hub_repository.org_hub_repo.id
+    name        = docker_hub_repository.org_hub_repo.name
+    namespace   = docker_hub_repository.org_hub_repo.namespace
+    description = docker_hub_repository.org_hub_repo.description
+  }
 }
 
+# Team outputs
 output "org_team_output" {
-  value = resource.docker_org_team.team
+  description = "Created team information"
+  value = {
+    id   = docker_org_team.team.id
+    name = docker_org_team.team.team_name
+  }
 }
 
-output "org_team_association_output" {
-  value = resource.docker_org_team_member.team_membership
+# Access token output
+output "access_token_uuid" {
+  description = "Created access token UUID"
+  value       = docker_access_token.access_token.uuid
+  sensitive   = true
 }
 
-output "access_tokens_uuids_output" {
-  value = resource.docker_access_token.access_token.uuid
+# === Security-Hardened Image Reference Output ===
+output "secure_image_example" {
+  description = "Demonstration of converting human-friendly tag to digest-pinned reference"
+  value = {
+    human_friendly = "alpine:latest"
```

Comment:
Consider retrieving these values dynamically (just to teach best-practices of the provider)